linked-container/src/test/java/com/example/linkedcontainer/RedmineContainer.java

```diff
@@ -0,0 +1,44 @@
+package com.example.linkedcontainer;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.traits.LinkableContainer;
+import org.testcontainers.containers.wait.Wait;
+
+/**
+ * A linkable Redmine container.
+ */
+public class RedmineContainer<SELF extends RedmineContainer<SELF>> extends GenericContainer<SELF> implements LinkableContainer {
+
+    private static final int REDMINE_PORT = 3000;
+
+    public RedmineContainer(String dockerImageName) {
+        super.setDockerImageName(dockerImageName);
+    }
+
+    @Override
+    protected Integer getLivenessCheckPort() {
+        return getMappedPort(REDMINE_PORT);
+    }
+
+    @Override
+    protected void configure() {
+        addExposedPort(REDMINE_PORT);
+        waitingFor(Wait.forHttp("/"));
+    }
+
+    @Override
+    public SELF withEnv(String key, String value) {
```

Comment:
If I remove this override then `withEnv` returns `GenericContainer` instead of `RedmineContainer`. The following line then gives an incompatible type error:
```
    private RedmineContainer redmineContainer = new RedmineContainer("redmine:3.3.2")
            .withLinkToContainer(postgreSQLContainer, "postgres")
            .withEnv("POSTGRES_ENV_POSTGRES_USER", POSTGRES_USERNAME)
            .withEnv("POSTGRES_ENV_POSTGRES_PASSWORD", POSTGRES_PASSWORD);
```
Is there another way to do this?